### PR TITLE
Fixed an issue where the assignment editor's save button would appear without making changes

### LINKED ISSUE #719: Format editor save button appears when clicking around on entrynodes
**Describe the bug**
Format editor save button appears when clicking around on entrynodes

**To Reproduce**
1. Go to an assignment editor page
2. Click some entrynodes without editing them
4. See save button appear

**Expected behavior**
No save button when you are not editing

### PR DESCRIPTION
**Pull request by Larspolo**
## Description
Fixes format editor save button appearing
Improved handling of target points

## Summary of changes
- Fixes #719

### CODE REVIEW

File: src/django/VLE/migrations/0030_tager_to_float.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Generated by Django 2.1.7 on 2019-09-22 12:14
```
Comment (mjvkeulen) [inline src/django/VLE/migrations/0030_tager_to_float.py]: tager_to_float?

File: src/django/VLE/serializers.py [hunk 2]
```diff
@@ -421,15 +421,15 @@ class Meta:
 
     def get_unlock_date(self, node):
         if node.type == Node.ENTRYDEADLINE and node.unlock_date is not None:
-            return node.unlock_date.strftime('%Y-%m-%d %H:%M')
+            return node.unlock_date.strftime('%Y-%m-%dT%H:%M')
```
Comment (engelhamer) [inline src/django/VLE/serializers.py]: How about we do this for all dates back end, so this data will always be uniform?

File: src/django/VLE/utils/generic_utils.py [hunk 3]
```diff
@@ -229,7 +232,12 @@ def update_presets(assignment, presets, new_ids):
         preset_node.lock_date = lock_date if lock_date else None
 
         if preset_node.type == Node.PROGRESS:
-            preset_node.target = target
+            if target > 0 and target <= assignment.points_possible:
+                preset_node.target = target
+            else:
+                raise VLEBadRequest(
+                    'Progress goal needs to be between 0 and the maximum amount for the assignment: {}'
```
Comment (engelhamer) [inline src/django/VLE/utils/generic_utils.py]: This is not enforced in the front-end. Would be nice to have a limit on the input field there.

File: src/django/VLE/serializers.py (hunk 2 above)
Comment (Larspolo) [reply src/django/VLE/serializers.py]: This is for all dates using the serializer. Default is already does it this way. Did I miss it anywhere?

Comment (engelhamer) [reply src/django/VLE/serializers.py]: Yes, course start / end dates and assignment unlock, due and lock.

File: src/vue/src/components/format/FormatPresetNodeCard.vue [hunk 4]
```diff
@@ -149,6 +149,7 @@
                 class="theme-input"
                 placeholder="Amount of points"
                 min="1"
+                :max="assignmentDetails.points_possible"
```
Comment (engelhamer) [inline src/vue/src/components/format/FormatPresetNodeCard.vue]: Maybe also add a check to the format validation to indicate that some preset nodes have a goal that is too high?

File: src/django/VLE/serializers.py (hunk 2 above)
Comment (Larspolo) [reply src/django/VLE/serializers.py]: image

And course start/end is only a date, so thats in the form of `%Y-%m-%d`
image

**Comment by mjvkeulen** [comment]
I like the split of the todo-square in needs marking and publish. I don't like seeing `0 | X` tho, would a single box with `X` not be prefered for this scenario?

**Pull request merged**